CONTRIBUTING.md

```diff
@@ -40,3 +49,105 @@ labels
 
 We also have a [slack channel](https://openssf.slack.com/archives/C03U677QD46)
 that you can reach us at and ask questions!
+
+## Contributor ladder
+
+We are excited that you want to contribute to GUAC! This contributor ladder
+outlines different contributor roles within the project, along with
+responsibilities and privileges that come with them.
+
+Since GUAC is a complex project, there are 5 topic areas of interest:
+
+- Ingestion: ingest software security metadata. Needs to write parsers for
+  documents, maintain ingestion logic, write new collectors, etc.
+- GraphQL: define the GraphQL interface used between ingestion pipeline and
+  backend, and between backend and GUAC-based applications and front-ends.
+- Backends: define efficient code to interface with database backends (Neo4j, etc.).
+- Front-end: GUAC visualisation libraries, Javascript, CSS
+- Clients: GUAC CLI, CI Checks, Policy Engines, IDE Plugins, etc.
+
+The contribution ladder is summarized in the table below:
+
+|         Role          | Responsibilities | Requirements | Privileges | GitHub access level |
+|-----------------------|------------------|--------------|------------|---------------------|
+| Community Participant | Follow [Google's Open Source Community Guidelines](https://opensource.google/conduct/) | N/A | N/A | Read (default) |
+| Reviewer              | Review contributions from other members | Prove technical expertise in at least one GUAC topic area | Can approve PRs | Write |
+| Owner                 | Set direction and priorities for topic area. Approval of contributions | Deep technical expertise in topic area. Proven record of reviews and authorship. | Slack channel invitation | Write |
+| Maintainer            | Set direction and priorities of project. Participate in weekly syncs | Expertise in 3 out of the 5 topic areas | (almost) complete repository access | Maintain |
+
+Each level of the ladder is also described in more details below.
+
+### Community Participant
+
+Each new contributor to GUAC starts at this level. There is no longer a formal
+requirement to [register desire to contribute in issue
+#1](https://github.com/guacsec/guac/issues/1).
+
+There are no requirements and no privileges. Every community participant can
+create issues, fork the repository, create PRs, add comments to issues and PRs.
+
+### Reviewer
+
+Once a community participant has demonstrated expertise in one GUAC topic area,
+owners and/on maintainers can propose graduating the community participant to
+the next role, that of Reviewer.
+
+A reviewer will be responsible of reviewing contributions from other
+contributors that touch the corresponding topic area. They should ensure that
+the PRs are adequately tested, follow the guidelines for the specific topic
+area.
+
+Since reviewer gain write access to the repository, they can also contribute
+directly, without needing to create a fork.
```

Comment:
hmm, i know probably not the best practice, but i keep doing it out of convenience. i think we should definitely still encourage folks to use forks so as not to have a danging branches problem (which CNCF TAG security repo suffers from).

Reply:
+1